modular_skyrat/modules/borg_buffs/code/robot.dm

```diff
@@ -116,3 +116,78 @@
 		tool_behaviour = TOOL_KNIFE
 		to_chat(user, span_notice("You attach the knife bit to [src]."))
 		icon_state = "knife_screw_cyborg"
+
+/obj/item/inducer/cyborg
+	name = "Cyborg Inducer"
+	desc = "A tool for inductively charging internal power cells using the battery of a cyborg"
+	powertransfer = 250
+
+
+
+/obj/item/inducer/cyborg/attackby(obj/item/W, mob/user)
+	if(W.tool_behaviour == TOOL_SCREWDRIVER)
+		return
+	if(istype(W, /obj/item/stock_parts/cell))
+		return
+
+/obj/item/inducer/cyborg/recharge(atom/movable/A, mob/user)
+	if(!iscyborg(user)) //
+		return
+	var/mob/living/silicon/robot/borgy = user
+	cell = borgy.cell //
+	if(!isturf(A) && user.loc == A)
+		return FALSE
+	if(recharging)
+		return TRUE
+	else
+		recharging = TRUE
+	var/obj/item/stock_parts/cell/C = A.get_cell()
+	var/obj/O
+	var/coefficient = 1
+	if(istype(A, /obj/item/gun/energy))
+		to_chat(user, span_alert("Error unable to interface with device."))
+		return FALSE
+	if(istype(A, /obj/item/clothing/suit/space))
+		to_chat(user, span_alert("Error unable to interface with device."))
+		return FALSE
+	if(cell.charge <= 1000 ) // Cyborg charge safety. Prevents a borg from inducing themself to death.
+		to_chat(user, span_alert("Unable to charge device. User battery safety engaged."))
+		return
+	if(istype(A, /obj))
+		O = A
+	if(C)
+		var/done_any = FALSE
+		if(C.charge >= C.maxcharge)
+			to_chat(user, span_notice("[A] is fully charged!"))
+			recharging = FALSE
+			return TRUE
+		user.visible_message(span_notice("[user] starts recharging [A] with [src]."), span_notice("You start recharging [A] with [src]."))
+		while(C.charge < C.maxcharge)
+			if(do_after(user, 10, target = user) && cell.charge)
```

Comment:
Could early return this

Reply:
You might also want to check if this only allows you to charge things when your charge falls under 1000 J after the first charge (I feel like it won't, which could be bad).
```suggestion
			if(do_after(user, 1 SECONDS, target = user) && cell.charge)
```